### ISSUE TITLE
Bug: duplicate outputs, even when no duplicates in dictionary

### ISSUE THREAD

**Issue by JellyJoe198** [JellyJoe198/Word-sorter#4]
Some words in the output are in the outputted twice, even though the input does not have duplicates.

```Python
wrdLength = 6 ## length of words to output
addOne = 1 ## how much longer than wrdLength will it also allow?
strictness = wrdLength -1  ## how many correct letters required (same as wrdLength)
## NOTE: for each blank, change strictness by -1

letters = ['e','i','r','c'] ## EeIMRC_

##if a letter is repeated, put its location in letters[] here (starts at 0)
repeats = [0] ## (e)

required = {
    0:  'm',
    1:  '',
    2:  '',
    3:  '',
    4:  '',
    5:  '',
    6:  '',
    7:  '',
    8:  '',
    9:  '',
    10: '',
    11: '',
    12: '',
    13: '' }
required_necessary = True   # False if other letters can fill a required spot
required_preffered = True   # BOTH False to disable `required` check entirely

willPrint = True ##print the words that it finds to console?
willScore = 1 ##will it find the score? (0=no, 1=print only, 2=print and write)
fileType = 'txt' # the file extension of Output (recommended: txt or csv)

#...#

tf = "dictionaries/filteredCombinedDict4a.txt" ## input dictionary file
## it also happened with words_alpha.txt but not as many words were duplicated
```

the output from words_alpha (* on the broken parts):
```
decree
deicer
feerie
fierce
greece
griece
keerie
mechir
meecher
meeker
*meered
*meered
meeter
meiler
*mercer
*mercer
*mercies
*mercies
*merice
*merice
metier
metric
micher
mincer
misere
morice
peerie
piecer
pierce
tierce

```

**Comment by JellyJoe198** [comment]
changing `required_preffered` to `False` did not change the output.
setting `required_necessary = False` and `required_preffered = True` is also broken

Also some of the duplicate words (but not all) have different scores:
```
mercer  —  11
mercer  —  12
mercies  —  12
mercies  —  12
merice  —  11
merice  —  12
```

**Comment by JellyJoe198** [comment]
fixed in 0.7.0
in line 141 the writeAnswer() command was on the inside `for letter in word`, so if the word got a score higher than required it would be run twice. Moved out of the for command
